src/libsync/theme.cpp

```diff
@@ -280,6 +280,85 @@ QIcon Theme::uiThemeIcon(const QString &iconName, bool uiHasDarkBg) const
     return QIcon(QPixmap(iconPath));
 }
 
+QPixmap Theme::pixmapForBackground(const QString &fileName, const QColor &backgroundColor) const
+{
+    Q_ASSERT(!fileName.isEmpty());
+
+    const auto iconBasePath = QStringLiteral(":/client/theme");
+
+    // some icons are present in white or black only, so, we need to check both when needed
+    const auto iconBaseColors = QStringList({ QStringLiteral("black"), QStringLiteral("white") });
+
+    const QString pixmapColor = backgroundColor.isValid() && !Theme::isDarkColor(backgroundColor) ? "black" : "white";
+
+    const QString cacheKey = fileName + QLatin1Char(',') + pixmapColor;
+
+    auto &cachedPixmap = _pixmapCache[cacheKey];
+
+    if (cachedPixmap.isNull()) {
+        if (iconBaseColors.contains(pixmapColor)) {
+            return cachedPixmap = QPixmap::fromImage(QImage(iconBasePath + QLatin1Char('/') + pixmapColor + QLatin1Char('/') + fileName));
+        }
+
+        const auto drawSvgWithCustomFillColor = [](const QString &sourceSvgPath, const QString &fillColor) {
+            QSvgRenderer svgRenderer;
+            
+            if (!svgRenderer.load(sourceSvgPath)) {
+                return QPixmap();
+            }
+
+            // render source image
+            QImage svgImage(svgRenderer.defaultSize(), QImage::Format_ARGB32);
```

Comment:
can you check what happen if the screen scale factor is `> 1` and we call `svgRender.defaultSize()` ?
we need to ensure that rendering size takes the HighDPI scale factor into account

Reply:
@mgallien Yes. Checked with 100%, 125% ad 150% of screen scale.

Reply:
do you think somebody should check on Linux or MacOS ?

Reply:
@mgallien I can check on Linux. Isn't `QSvgRenderer `cross-platform anyway?

Reply:
tests show that scaling is being done correctly on all platforms